gammapy/cube/make.py

```diff
@@ -257,3 +260,125 @@ def _check_selection(selection):
             raise ValueError("Selection not available: {!r}".format(name))
 
     return selection
+
+class IrfMapMaker:
+    """Make IRF maps from IACT observations.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.WcsGeom`
+        Reference map geometry with true energy axis
+    offset_max : `~astropy.coordinates.Angle`
+        Maximum offset angle
+    rad_axis : `~gammapy.maps.MapAxis`
+        The rad axis of the PsfMap to be built
+    migra_axis: `~gammapy.maps.MapAxis`
+        The migra axis of the EDispMap to be built
+    """
+
+    def __init__(self, geom, offset_max, rad_axis=None, migra_axis=None):
+        if not isinstance(geom, WcsGeom):
+            raise ValueError("IrfMapMaker only works with WcsGeom")
+
+        if geom.is_image:
+            raise ValueError("IrfMapMaker only works with geom with an energy axis")
+
+        self.geom = geom
+        self.offset_max = Angle(offset_max)
+
+        if rad_axis is None and migra_axis is None:
```

Comment:
Why not just make `rad_axis` and `migra_axis` required arguments? The current design does not make sense
 to me: declaring both arguments as optional and raising an error if they are not specified. Or do you plan to implement a default for `migra_axis` and `rad_axis` later?

Reply:
The point here is that you want the `IRFMapMaker` to be able to make only the `PSF` or the `EDisp`. It is useless to pass a list or `Map` types to be produced as in `MapMaker`, the simple list of extra axes already provides the information needed.

Reply:
OK, thanks for the explanation, I see your point. Maybe add a short comment to the docstring?